backend-services/autoignore/core/autoignore.py

```diff
@@ -120,9 +120,34 @@ class ConfigHandler(RequestHandler):
     def initialize(self, shared_memory_manager_dict):
         self.shared_memory_manager_dict = shared_memory_manager_dict
 
+    def get(self):
+        """
+        Provides current configuration primitives (in the form of a JSON dict) to the requester.
+        Format:
+        {
+            "autoignore_rules": <dict>,
+            "config_timestamp": <timestamp>
+        }
+        """
+        ret_dict = {}
+
+        shared_memory_locks["autoignore"].acquire()
```

Comment:
I don't think we need to lock for READ-ONLY operations

Reply:
we need those since we deal with dicts etc. which may be under change at the time. The locks ensure us that this is the current complete state of the copy we pass between the processes.

Reply:
I will remove as discussed, good point!

Reply:
I removed all the locks in latest commit. Please check and let me know if you have further comments.